### ISSUE TITLE
 Overloads with Parameters of type `Type[Any]`  are overlapped by overloads with parameters of Type `Any`

### ISSUE THREAD

**Issue by MHDante** [microsoft/pyright#3329]
**Describe the bug**
Overloads with Parameters of type `Type[Any]`  are overlapped by overloads with parameters of Type `Any`

**To Reproduce**
Copy code below. 
Code produces error:
```
error: Overload 4 for "Make" will never be used because its parameters overlap overload 3 (reportOverlappingOverload)
```

**Expected behavior**

Code below should produce no errors.

**Screenshots or Code**

``` python

class Maker:
  @overload
  def Make(self, toCopy:str) -> str: ...
  @overload
  def Make(self, toCopy:int) -> int: ...
  @overload
  def Make(self, toCopy:Type[Maker]) -> Maker: ...
  @overload
  def Make(self, toCopy:Type[Any]) -> bool | Maker: ...
  @overload
  def Make(self, toCopy:Any) -> str | int | bool | Maker: ...
  # ^^ last overload causes reportOverlappingOverload

```

![image](https://user-images.githubusercontent.com/4652122/162659565-bfb174f5-f9e7-4d03-96e4-b329207cf495.png)
![image](https://user-images.githubusercontent.com/4652122/162659610-44ec77d0-e3d7-42ae-a311-2541a73b6fff.png)

**VS Code extension or command-line**
pyright 1.1.236
![image](https://user-images.githubusercontent.com/4652122/162659643-462fff8c-51e8-41b6-87dd-5398f087c726.png)

![image](https://user-images.githubusercontent.com/4652122/162659700-9a9dd4d0-1722-4fdd-a4ad-407f5375454c.png)
pylance v2022.4.0

**Additional Info**

Interestingly, type hints still work correctly
![image](https://user-images.githubusercontent.com/4652122/162660570-0047108d-5c16-4948-bedd-0bdbab9b5cc6.png)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.237, which I just published. It will also be included in the next release of pylance.
